packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -338,6 +349,12 @@ class CupertinoSlidingSegmentedControl<T extends Object> extends StatefulWidget
   /// The map must have more than one entry.
   final Map<T, Widget> children;
 
+  /// The identifying keys and the corresponding widget status whether the
+  /// segment is enabled or not.
+  ///
+  /// If this is null, all segments are selectable by default.
+  final Map<T, bool>? setEnabled;
```

Comment:
Hmm, I would say making `children` a `Map<T, Widget>` was a bit questionable, now we have to introduce a separate data structure as part of the API to track the enabled status for each child. Still I think it would be better to use a Set (and make it not nullable) if we can't change the `children` API.

Reply:
`setEnable` => `disabledChildren`?

Reply:
Document how this interacts with `groupValue`?